File: backend/app/models/project.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey, Enum
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import Base
import enum

class DocumentType(str, enum.Enum):
    DOCX = "docx"
    PPTX = "pptx"

class Project(Base):
    __tablename__ = "projects"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String, nullable=False)
    topic = Column(String, nullable=False)
    document_type = Column(Enum(DocumentType), nullable=False)
    
    # Generated content
    generated_content = Column(Text, nullable=True)  # JSON string for structured content
    
    # Metadata
    owner_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    
    # Timestamps
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    owner = relationship("User", back_populates="projects")

File: frontend/src/types/project.ts
export interface Project {
  id: number
  title: string
  topic: string
  document_type: 'docx' | 'pptx'
  generated_content?: string
  owner_id: number
  created_at: string
  updated_at?: string
}

export interface ProjectCreate {
  title: string
  topic: string
  document_type: 'docx' | 'pptx'
}

